sql/src/main/java/io/crate/execution/engine/collect/files/FileReadingIterator.java

```diff
@@ -201,18 +201,27 @@ private void advanceToNextFileInput() throws IOException {
         Predicate<URI> uriPredicate = generateUriPredicate(fileInput, fileUri.globPredicate);
         List<URI> uris = getUris(fileInput, fileUri.uri, fileUri.preGlobUri, uriPredicate);
 
-        if (uris.size() > 0) {
+        if (uris.isEmpty()) {
+            LOGGER.warn("No files matched the pattern '{}'", fileUri.uri);
+        } else {
             currentInputIterator = uris.iterator();
             advanceToNextUri(fileInput);
         }
     }
 
     private void initCurrentReader(FileInput fileInput, URI uri) throws IOException {
-        InputStream stream = fileInput.getStream(uri);
-        if (stream != null) {
-            currentReader = createBufferedReader(stream);
-            currentLineNumber = 0;
-            lineProcessor.readFirstLine(currentUri, inputFormat, currentReader);
+        try {
+            InputStream stream = fileInput.getStream(uri);
+            if (stream != null) {
+                currentReader = createBufferedReader(stream);
+                currentLineNumber = 0;
+                lineProcessor.readFirstLine(currentUri, inputFormat, currentReader);
+            } else {
+                LOGGER.warn("Couldn't open reader for URI '{}'", uri);
+            }
+        } catch (IOException e) {
+            LOGGER.warn("{} when reading from URI '{}': {}",
```

Comment:
why not logging the complete exception? wouldn't that help in many cases?

Reply:
I wanted to keep this readable and avoid spaming the log with the full stack trace. I don't think the full exception really helps. The type and the message should be enough.

Reply:
Maybe we could log the full exception with debug flag.

Reply:
I fine with your solution, I wouldn't add extra complexity (branches) just for that.

Reply:
OK, so I'll keep it as it is?